src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -260,7 +260,12 @@ protected Object invokeStep(StepDescriptor d, String name, Object args) {
             }
             d.checkContextAvailability(context);
             Thread.currentThread().setContextClassLoader(CpsVmExecutorService.ORIGINAL_CONTEXT_CLASS_LOADER.get());
-            s = d.newInstance(ps.namedArgs);
+            if (Util.isOverridden(StepDescriptor.class, d.getClass(), "newInstance", Map.class)) {
+                s = d.newInstance(ps.namedArgs);
+            } else {
+                DescribableModel<?> stepModel = DescribableModel.of(d.clazz);
```

Comment:
nit: I _think_ the type parameters would be ok if you did the following, and then you wouldn't need the explicit cast to `Step` on the next line.
```suggestion
                DescribableModel<? extends Step> stepModel = DescribableModel.of(d.clazz);
```